bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/CallingUserActionSequenceElement.java

```diff
@@ -56,12 +58,37 @@ public AbstractActionSequenceElement<EntryLevelSystemCall> evaluateDataFlow(List
                 .flatMap(it -> it.getVariableCharacterisation_VariableUsage()
                         .stream())
                     .collect(Collectors.toList());
+    	
+    	if (this.isCalling()) {
+        	checkParameter(variableCharacterisations);
+        }
+    	
 
         PCMDataCharacteristicsCalculator characteristicsCalculator = new PCMDataCharacteristicsCalculator(new ArrayList<>(variables), nodeCharacteristics);
         variableCharacterisations.stream()
             .forEach(it -> characteristicsCalculator.evaluate(it));
        return new CallingUserActionSequenceElement(this, characteristicsCalculator.getCalculatedCharacteristics(), nodeCharacteristics);
     }
+    
+    private void checkParameter(List<VariableCharacterisation> variableCharacterisations) {
+    	List<String> parameter = 
+    			this.getElement().getOperationSignature__EntryLevelSystemCall().getParameters__OperationSignature().stream()
+    			.map(it -> it.getParameterName())
+    			.toList();
+    	
+    	List<String> referencedParameter =
+    			variableCharacterisations.stream()
+    			.map(it -> it.getVariableUsage_VariableCharacterisation().getNamedReference__VariableUsage().getReferenceName())
+    			.toList();
+    	
+    	referencedParameter.stream()
+    	.filter(it -> !parameter.contains(it))
+    	.forEach(it -> {
+    		logger.warn("Unknown reference to variable " + it + " in variable characterisation in element " + this.toString());
+    		logger.warn("Present variables:" + parameter + ", Referenced parameter: " + referencedParameter);
+    		throw new IllegalStateException();
```

Comment:
Why does this throw an exception but the upper implementation of ExternalCalls does not?

Reply:
This has left to debug the warnings, but I forgot to remove it. I've removed it now